src/cpp/main.cpp

```diff
@@ -0,0 +1,108 @@
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
+extern "C" {
+  #include "config.h"
+  #include "data.h"
+  #include "functions.h"
+}
+
+// Simulation structures.
+Particle *particles;
+ParticleProperties *properties;
+Contact *contacts_buffer;
+double *normal_forces;
+double *tangent_forces;
+Vector *forces;
+Vector *accelerations;
+Vector *velocities;
+Vector *displacements;
+
+/**
+ * Initialize all simulation structures according to the number of particles.
+ * The structures are effectively initialized with zeros.
+ */
+void initialize(const Config *config) {
+  // The number of particles is equals to the simulation size,
+  // plus the falling particle (the first one).
+  size_t num_particles = config->simulation_size + 1;
+
+  // Allocate memory.
+  particles = (Particle*) calloc(num_particles, sizeof(Particle));
+  properties = (ParticleProperties*) calloc(num_particles, sizeof(ParticleProperties));
+  contacts_buffer = (Contact*) calloc(size_triangular_matrix(num_particles), sizeof(Contact));
+  normal_forces = (double*) calloc(num_particles * num_particles, sizeof(double));
+  tangent_forces = (double*) calloc(num_particles * num_particles, sizeof(double));
+  forces = (Vector*) calloc(num_particles, sizeof(Vector));
+  accelerations = (Vector*) calloc(num_particles, sizeof(Vector));
+  velocities = (Vector*) calloc(num_particles, sizeof(Vector));
+  displacements = (Vector*) calloc(num_particles, sizeof(Vector));
+
+  // Initialize the structures.
+  for (size_t i = 0; i < num_particles; ++i) {
+    particles[i].radious = config->r;
```

Comment:
@ibotero2 How should we initialize the `x_coordinate`  `y_coordinate` of the particles _(including the falling one)_?